lib/Tools/LinearLayout.cpp

```diff
@@ -1338,4 +1337,21 @@ std::string ColumnAction::toString() const {
   return ret;
 }
 
+PaddedLinearLayout::PaddedLinearLayout(LinearLayout linear,
+                                       ArrayRef<unsigned> intervals,
+                                       ArrayRef<unsigned> paddings)
+    : linear(std::move(linear)) {
```

Comment:
Isn't this linear layout always the identity Layout? As such, I don't think we need to save it. The intervals and paddings should be enough to define the layout.
Also, since it is the identity, now when you do `distributed.invertAndCompose(padded.getLayout())` you could simply consider `distributed` and move on.

Reply:
Yes it will always be identity when the attribute is indeed a `PaddedSharedEncodingAttr`. Though this utility class is also meant to be able to handle non-padded shared layout so it works with the same code path--e.g., for swizzled layout we will use this linear layout part to hold it so it can still work with `invertAndCompose`; just that the intervals and paddings are empty this time. Also this way we can provide a `ttg::toPaddedLinearLayout` that works on all shared layout like `ttg::toLinearLayout`. I think this makes the implementation more consistent. But let me know if you think otherwise. :)

Reply:
Is there any use case where we would want to have swizzling and padding? If not, I think it'd be better to keep things simple and assume the internal layout is just the identity.

Reply:
Currently the name is reflecting what's inside the struct. I changed the name to reflect what it's intended for with 2de5b2f and improved comments a bit with b5f258e. Let me know how you feel now. If it's still confusing, I'll revert to do something like https://github.com/antiagainst/triton/commit/61e11fc8266025a627bacd0bee7217bd2547a844.

Reply:
Sure thing. Reverted current impl and landed 8fa8d8db1d198babe4675162d4119e24dff10136.